app/EventHandlers/TrackBannerDonationRedirects.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\App\EventHandlers;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class TrackBannerDonationRedirects implements EventSubscriberInterface {
+
+	public const PIWIK_CAMPAIGN = 'piwik_campaign';
+	public const PIWIK_KWD = 'piwik_kwd';
+
+	private SessionInterface $session;
+	private string $submissionRoute;
+	private string $confirmationRoute;
+	private string $bannerSubmissionUrlParameter;
+
+	public static function getSubscribedEvents() {
+		return [
+			KernelEvents::REQUEST => 'onKernelRequest',
+			KernelEvents::RESPONSE => 'onKernelResponse'
+		];
+	}
+
+	public function __construct( SessionInterface $session, string $submissionRoute, string $confirmationRoute, string $bannerSubmissionUrlParameter ) {
+		$this->session = $session;
+		$this->submissionRoute = $submissionRoute;
+		$this->confirmationRoute = $confirmationRoute;
+		$this->bannerSubmissionUrlParameter = $bannerSubmissionUrlParameter;
```

Comment:
```suggestion
		$this->isBannerSubmissionUrlParameter = $bannerSubmissionUrlParameter;
```
Just a suggestion for naming this thing better - it's an an URL parameter that indicates that the submission comes from a banner.

Reply:
That variable contains the name of the parameter though, rather than the value